api/api/src/main/scala/uk/ac/wellcome/platform/api/rest/ImagesController.scala

```diff
@@ -0,0 +1,50 @@
+package uk.ac.wellcome.platform.api.rest
+
+import akka.http.scaladsl.server.Route
+import com.sksamuel.elastic4s.{ElasticClient, Index}
+import uk.ac.wellcome.display.models.v2._
+import uk.ac.wellcome.display.models.Implicits._
+import uk.ac.wellcome.elasticsearch.ElasticConfig
+import uk.ac.wellcome.platform.api.Tracing
+import uk.ac.wellcome.platform.api.models.ApiConfig
+import uk.ac.wellcome.platform.api.services.{
+  ElasticsearchService,
+  ImagesService
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ImagesController(
+  elasticClient: ElasticClient,
+  implicit val apiConfig: ApiConfig,
+  elasticConfig: ElasticConfig)(implicit ec: ExecutionContext)
+    extends CustomDirectives
+    with Tracing {
+
+  def singleImage(id: String, params: SingleImageParams): Route =
+    getWithFuture {
+      transactFuture("GET /images/{imageId}") {
+        val index =
+          params._index.map(Index(_)).getOrElse(elasticConfig.imagesIndex)
+        imagesService
+          .findImageById(id)(index)
+          .flatMap {
+            case Right(Some(image)) =>
+              Future.successful(
+                complete(
+                  ResultResponse(
+                    context = contextUri,
+                    result = DisplayImage(image)
+                  )
+                )
+              )
+            case Right(None) =>
+              Future.successful(notFound(s"Image not found for identifier $id"))
+            case Left(err) => Future.successful(elasticError(err))
```

Comment:
I don't think this needs to be `flatMap` rather than just a `map`? Everything is just returning a `Future.successful(...)`.

Reply:
Good point, I'd copied this mindlessly from the WorksController where it can be changed too!